### DISCUSSION TITLE
테스트 코드에서 빌더의 줄바꿈은 어떻게 할까요?!

### DISCUSSION THREAD

**Discussion by Kim0914** [woowacourse-teams/2023-pium#239] [질문있어요] [answered]
**현재 프로덕션 코드에서는 빌더의 줄바꿈을 다음과 같이 `한 줄에 점 하나` 형식으로 가져갑니다.**

```java
PetPlant.builder()
                .dictionaryPlant(dictionaryPlant)
                .member(member)
                .nickname(request.getNickname())
                .imageUrl(dictionaryPlant.getImageUrl())
                .location(request.getLocation())
                .flowerpot(request.getFlowerpot())
                .light(request.getLight())
                .wind(request.getWind())
                .birthDate(request.getBirthDate())
                .lastWaterDate(request.getLastWaterDate())
                .waterCycle(request.getWaterCycle())
                .nextWaterDate(LocalDate.now().plus(request.getWaterCycle(), ChronoUnit.DAYS))
                .build();
```

현재 테스트 코드에서 builder를 사용하는 컨벤션이 섞여 있는 것 같아요 🤓

![image](https://github.com/woowacourse-teams/2023-pium/assets/68818952/d5afddac-4d10-46bc-87d8-b8de1ff1d7f4)
![image](https://github.com/woowacourse-teams/2023-pium/assets/68818952/76d6d125-b27b-4a6a-be15-a14129c670d4)
  
  
저는 테스트 코드에서는 `한 줄에 점 하나`가 크게 중요하지 않은 것 같아서, 한 줄에 작성해도 되지 않을까 생각합니다 !
  
  

다만, 아래와 같이 픽스쳐를 사용하기 애매(예외를 발생시키기 위해 특정 필드에 값을 채워함)해서  체이닝을 한 줄에 표현할 수 없는 경우에는 줄 바꿈을 가져가는게 좋을 것 같아요.
![image](https://github.com/woowacourse-teams/2023-pium/assets/68818952/44e913a2-cd14-4051-b3b6-3058e8468d38)

이 부분을 맞춰봤으면 좋겠는데 다른 분들의 생각이 궁금합니다 !

**Comment by Choi-JJunho** [comment]
저는 그레이 의견에 동의합니다~

정리하면 다음과 같은 의견입니다.

테스트코드에서 builder를 사용할 경우 기본적으로 한줄로 작성한다.
테스트 시나리오 상 객체의 내부 값을 제어(선언 및 참조)하고 활용해야하는 경우는 엔터로 구분하여 작성한다.

**Comment by yeonkkk** [comment]
**저는 `통일성`과 `가독성` 측면에서 모두 동일하게 `한줄에 점 하나`로 가져가는 것도 좋을 것 같습니다.**

테스트 코드는 전체 로직를 이해하기 위한 명세와 같은 역할을 하기 때문에, 테스트 코드 내에서도 가독성을 고려하면 좋을 것 같아서요!
그리고 `체이닝을 한 줄에 표현할 수 없는 경우` 가 명확한 기준이 되기 어려울 수도 있을 것 같아요. 
그렇게 되면 각자의 기준대로 사용해서 지금 상황과 유의미한 차이를 얻기는 힘들 것 같아요.

사실 정답은 없는 문제라서 그레이의 의견대로 해도, 어느 한 쪽으로 통일해도 괜찮긴 합니다 ㅎㅎ🙂

**Comment by rawfishthelgh** [comment, marked as answer]
저는 주노, 그레이의 의견과 동일합니다! 
테스트의 빌더 사용 목적이 단순 객체의 생성 측면에 있고, 객체 내부의 필드값 제어가 아니라면 빌더의 가독성은 크게 중요하지 않은 것 같아요.

게다가 오히려 한줄로 작성한 빌더와, 줄바꿈을 추가한 빌더를 구분한다면 테스트 의도 파악이 더 쉽지 않을까 생각이 듭니다.
